tests/python_frontend/indirections_test.py

```diff
@@ -0,0 +1,408 @@
+    assert (np.allclose(res, A[x, y]))
+
+
+@dc.program
+def indirection_array_op(A: dc.float32[10], x: dc.int32[10]):
+    i = 0
+    return A[x[i] + 2]
+
+
+def test_indirection_array_op():
+    A = np.random.randn(10).astype(np.float32)
+    x = np.random.randint(0, 8, size=(10,), dtype=np.int32)
+    res = indirection_array_op(A, x)[0]
+    assert (res == A[x[0] + 2])
+
+
+@dc.program
+def indirection_array_op_nsdfg(A: dc.float32[10], x: dc.int32[10]):
+    B = np.empty_like(A)
+    # TODO: This doesn't work with 0:A.shape[0]
+    for i in dc.map[0:10]:
+        B[i] = A[x[i] + 5]
+    return B
+
+
+def test_indirection_array_op_nsdfg():
+    A = np.random.randn(10).astype(np.float32)
+    x = np.random.randint(0, 5, size=(10,), dtype=np.int32)
+    res = indirection_array_op_nsdfg(A, x)
+    assert (np.allclose(res, A[x + 5]))
+
+
+@dc.program
+def indirection_array_range(A: dc.float32[10], x: dc.int32[10]):
+    i = 5
+    return np.sum(A[x[i]:x[i]+1])
+
+
+def test_indirection_array_range():
+    A = np.random.randn(10).astype(np.float32)
+    x = np.random.randint(0, 9, size=(10,), dtype=np.int32)
+    res = indirection_array_range(A, x)[0]
+    assert (np.allclose(res, np.sum(A[x[5]:x[5]+1])))
+
+
+@dc.program
+def indirection_array_range_nsdfg(A: dc.float32[10], x: dc.int32[10]):
+    B = np.empty_like(A)
+    # TODO: This doesn't work with 0:A.shape[0]
+    for i in dc.map[0:10]:
+        B[i] = np.sum(A[x[i]:x[i]+1])
+    return B
+
+
+def test_indirection_array_range_nsdfg():
+    A = np.random.randn(10).astype(np.float32)
+    x = np.random.randint(0, 9, size=(10,), dtype=np.int32)
+    res = indirection_array_range_nsdfg(A, x)
+    assert (np.allclose(res, A[x]))
+
+
+@dc.program
+def indirection_array_nested(A: dc.float32[10], x: dc.int32[10]):
+    i = 0
+    return A[x[x[i]]]
+
+
+def test_indirection_array_nested():
+    A = np.random.randn(10).astype(np.float32)
+    x = np.random.randint(0, 10, size=(10,), dtype=np.int32)
+    res = indirection_array_nested(A, x)[0]
+    assert (res == A[x[x[0]]])
+
+
+@dc.program
+def indirection_array_nested_nsdfg(A: dc.float32[10], x: dc.int32[10]):
+    B = np.empty_like(A)
+    # TODO: This doesn't work with 0:A.shape[0]
+    for i in dc.map[0:10]:
+        B[i] = A[x[x[i]]]
+    return B
+
+
+def test_indirection_array_nested_nsdfg():
+    A = np.random.randn(10).astype(np.float32)
+    x = np.random.randint(0, 10, size=(10,), dtype=np.int32)
+    res = indirection_array_nested_nsdfg(A, x)
+    assert (np.allclose(res, A[x[x]]))
+
+
+@dc.program
+def spmv(A_row, A_col, A_val, x):
+    y = np.empty(A_row.size - 1, A_val.dtype)
+
+    for i in range(A_row.size - 1):
+        cols = A_col[A_row[i]:A_row[i + 1]]
+        vals = A_val[A_row[i]:A_row[i + 1]]
+        y[i] = vals @ x[cols]
+
+    return y
+
+
+def test_spmv():
+
+    with dc.config.set_temporary('compiler',
+                                 'allow_view_arguments',
+                                 value=True):
```

Comment:
Why is this needed?

Reply:
Because the default value of this new compilation flag breaks the test unless I make explicit copies of all the data, which is unnecessary and undesirable in this case.